### DISCUSSION TITLE
make check error while building libmesh

### DISCUSSION THREAD

**Discussion by aeronova** [libMesh/libmesh#3506] [General]
Hello,

I'm trying to build the git version of libmesh using petsc-3.18.5 and slepc-3.18.3 on a Mac. I used the following for configuration:

./configure --prefix=/Users/aeronova/Development/local/lib/libmesh/git --enable-everything --with-methods="dbg"

The make process went smooth, but I get an error as below when I ran "make check."

Making check in examples
Making check in introduction/introduction_ex1
/Applications/Xcode.app/Contents/Developer/usr/bin/make  example-dbg     run.sh
  CXX      example_dbg-introduction_ex1.o
  CXXLD    example-dbg
make[3]: Nothing to be done for `run.sh'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
***************************************************************
* Running Example introduction_ex1:
*   ./example-dbg -d 3 /Users/aeronova/Development/local/src/libmesh-git/reference_elements/3D/one_hex27.xda 
***************************************************************
 
PETSC ERROR: Logging has not been enabled.
You might have forgotten to call PetscInitialize().
application called MPI_Abort(MPI_COMM_WORLD, 56) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=56
:
system msg for write_line failure : Bad file descriptor
FAIL: run.sh

I'd appreciate it if anyone would give me some tips to fix this error.

**Comment by jwpeterson** [comment]
Based on just this error message, it's difficult to say for sure what the issue might be. One guess is that you might have multiple MPIs installed on this machine, and PETSc is built with a different MPI than libmesh somehow. It should not be possible for that to happen in theory though, so it's just a guess.

**Reply by aeronova** [reply]
Thank you for the reply. I have removed mpich, petsc, and slepc installed by macports, but still got the same error. I am not sure of what files to see to get some clues to fix the error.

**Reply by roystgnr** [reply]
I'd look at your config.log, and your configure output if you've saved it.  If you've removed PETSc and you're still seeing error messages from PETSc then you haven't really removed PETSc.  And if there's a chance you've got multiple different PETSc installs visible in your environment that might lead to all sorts of disaster.
